src/Compilers/CSharp/Test/CommandLine/CommandLineTests.cs

```diff
@@ -8845,6 +8867,186 @@ public void Version()
             }
         }
 
+        [Fact]
+        public void RefOut()
+        {
+            var dir = Temp.CreateDirectory();
+            dir.CreateDirectory("ref");
+
+            var src = dir.CreateFile("a.cs");
+            src.WriteAllText(@"
+class C
+{
+    /// <summary>Main method</summary>
+    public static void Main()
+    {
+        System.Console.Write(""Hello"");
+    }
+}");
+
+            var outWriter = new StringWriter(CultureInfo.InvariantCulture);
+            var csc = new MockCSharpCompiler(null, dir.Path,
+                new[] { "/nologo", "/out:a.exe", "/refout:ref/a.dll", "/doc:doc.xml", "/deterministic", "a.cs" });
+
+            int exitCode = csc.Run(outWriter);
+            Assert.Equal(0, exitCode);
+
+            var exe = Path.Combine(dir.Path, "a.exe");
+            Assert.True(File.Exists(exe));
+
+            VerifyPEMetadata(File.OpenRead(exe),
+                new[] { "TypeDef:<Module>", "TypeDef:C" },
+                new[] { "MethodDef: Void Main()", "MethodDef: Void .ctor()" },
+                new[] { "CompilationRelaxationsAttribute", "RuntimeCompatibilityAttribute", "DebuggableAttribute" }
+                );
+
+            var doc = Path.Combine(dir.Path, "doc.xml");
+            Assert.True(File.Exists(doc));
+
+            using (var reader = new StreamReader(doc))
+            {
+                var content = reader.ReadToEnd();
+                var expectedDoc =
+@"<?xml version=""1.0""?>
+<doc>
+    <assembly>
+        <name>a</name>
+    </assembly>
+    <members>
+        <member name=""M:C.Main"">
+            <summary>Main method</summary>
+        </member>
+    </members>
+</doc>";
+                Assert.Equal(expectedDoc, content.Trim());
+            }
+
+            var output = ProcessUtilities.RunAndGetOutput(exe, startFolder: dir.ToString());
+            Assert.Equal("Hello", output.Trim());
+
+            var refDll = Path.Combine(dir.Path, "ref\\a.dll");
+            Assert.True(File.Exists(refDll));
+
+            VerifyPEMetadata(File.OpenRead(refDll),
+                new[] { "TypeDef:<Module>", "TypeDef:C" },
+                new[] { "MethodDef: Void Main()", "MethodDef: Void .ctor()" },
+                new[] { "CompilationRelaxationsAttribute", "RuntimeCompatibilityAttribute", "DebuggableAttribute" }
+                );
+
+            output = ProcessUtilities.RunAndGetOutput(refDll, startFolder: dir.ToString(), expectedRetCode: -532462766);
+
+            // Clean up temp files
+            CleanupAllGeneratedFiles(dir.Path);
```

Comment:
Good point. I wasn't disposing stream above (`File.OpenRead(refDll)`) so was probably leaking files.
Thanks

---
In reply to: [106547188](https://github.com/dotnet/roslyn/pull/17558#discussion_r106547188) [](ancestors = 106547188)